packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -49,13 +49,14 @@ const usePermissions = <Permissions = any, Error = any>(
         queryParams
     );
 
-    return useMemo(() => {
-        return {
+    return useMemo(
+        () => ({
             permissions: result.data,
             isLoading: result.isLoading,
             error: result.error,
-        };
-    }, [result]);
+        }),
```

Comment:
Same here with `permissions`?

Reply:
I'm pretty sure TypeScript complains that `permissions` may be null even after checking `isLoading` and `error`.

But improving the return type of `usePermissions` is outside the scope of this PR.

I only changed the syntax slightly for my eye pleasure on this file.